chromium/background.js

```diff
@@ -660,5 +682,57 @@ chrome.runtime.onMessage.addListener(function(message, sender, sendResponse){
     updateState();
   } else if (message.type == "remove_rule") {
     removeRule(message.object);
+  } else if (message.type == "import_settings") {
+    // This is used when importing settings from the options ui
+    import_settings(message.object).then(() => {
+      sendResponse(true);
+    });
   }
 });
+
+// Send a message to the embedded webextension bootstrap.js to get settings to import
+chrome.runtime.sendMessage("import-legacy-data", import_settings);
+
+/**
+ * Import extension settings (custom rulesets, ruleset toggles, globals) from an object
+ * @param settings the settings object
+ */
+async function import_settings(settings){
+  if(settings.changed){
+    // Load custom rulesets and add to storage
+    await new Promise(resolve => {
+      storage.set({"legacy_custom_rulesets": settings.custom_rulesets}, resolve);
+    });
+    load_legacy_custom_rulesets(settings.custom_rulesets);
+
+    // Load all the ruleset toggles into memory and store
+    let rule_toggle_promises = [];
+    for(let ruleset_name in settings.rule_toggle){
+      localStorage[ruleset_name] = settings.rule_toggle[ruleset_name];
+    }
+    (ruleset_name, ruleset_active => {
+      for(let host in all_rules.targets){
+        for(let ruleset of all_rules.targets[host]){
+          if(ruleset.name == ruleset_name){
+            ruleset.active = ruleset_active;
+            return;
+          }
+        }
+      }
+    })(ruleset_name, settings.rule_toggle[ruleset_name]);
+
+    // This line is necessary to clear cached lookups of rulesets already loaded
+    all_rules.ruleCache = new Map();
```

Comment:
We could do

```js
all_rules = new RuleSet(localStorage);
all_rules.addFromXml(loadExtensionFile('rules/default.rulesets', 'xml'));
load_legacy_custom_rulesets(settings.custom_rulesets);
```

Which is probably fine for now, but seems like we may run into complications when we are working on merging `sign-rulesets`.  Though I can take care of that problem when we come to it, I suppose.